### PR TITLE
Add transaction_from_json helper function and refactor StarkNet pallet tests

### PR DESCRIPTION
**Pull request by Matth26**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [x] Testing
- [ ] Other (please describe):

# What is the current behavior?

While writing tests for the starknet pallet, new transactions have to be manually created by using its constructor Transaction::new. This is very cumbersome and affects tests' readability overall.

Issue Number: #111 

# What is the new behavior?

- A new helper function, `transaction_from_json`, has been added. This function allows users to create Transaction objects directly from json files. 
- json transaction files are now stored in a dedicated `/resources/transactions` subfolder. 

# Does this introduce a breaking change?

- [ ] Yes
- [x] No

# Other information

All elements of the Transaction come from the json except for the `pub contract_class: Option<ContractClassWrapper>` field. 

The existing tests use a file from the blockifier crate, imported with the `include_bytes!` macro. So I opted to exclude it from the json and incorporate it as a function argument instead:

```Rust
pub fn transaction_from_json(json_str: &str, contract_content: &'static [u8])
```

If `contract_content` is set to `&[]`, the `contract_class` field will be set to `None`. Otherwise, we create the `ContractClassWrapper` from the `contract_content`.

### CODE REVIEW

**Review by EvolveArt — CHANGES_REQUESTED** [review]
Thanks for this! A few changes I'd like to see tho : 
- You could put the code from `transaction_from_json` in the `starknet_serde` primitive module.
- No need to rewrite all the tests, just modifying the existing ones to use the helper function will be fine
- I would like to see tests only on serialization/derserialization. Edge cases where it fails should be covered there.
- Finally would be great to have more explicit names about what the transactions actually do instead of referring to test names

File: crates/primitives/starknet/src/starknet_serde/mod.rs [hunk 1]
```diff
@@ -1 +1,321 @@
...
+use blockifier::test_utils::get_contract_class;
+use frame_support::BoundedVec;
+use hex::{FromHex, FromHexError};
+use serde::{Deserialize, Serialize};
+use sp_core::{H256, U256};
+
+use crate::execution::{CallEntryPointWrapper, ContractAddressWrapper, ContractClassWrapper, EntryPointTypeWrapper};
+use crate::transaction::types::{EventWrapper, MaxArraySize, Transaction};
+
+fn remove_prefix(input: &str) -> &str {
```
Comment (abdelhamidbakhta) [inline crates/primitives/starknet/src/starknet_serde/mod.rs:15]: please add rust doc

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
left few comments

**Review by EvolveArt — APPROVED** [review]
lgtm

**Review by abdelhamidbakhta — APPROVED** [review]
